src/transformers/models/test.py

```diff
@@ -0,0 +1,8 @@
+from wav2vec2 import Wav2Vec2 , Wav2Vec2Config
```

Comment:
Looks like there's an erroneous file here!

Reply:
Hello @sanchit-gandhi  i join below the code that I took directly on the documentation of Huggingface for the pretraining, but this returns Nan as loss  : 

```py
import torch
from transformers import AutoFeatureExtractor, Wav2Vec2ForPreTraining
from transformers.models.wav2vec2.modeling_wav2vec2 import _compute_mask_indices
from datasets import load_dataset

feature_extractor = AutoFeatureExtractor.from_pretrained("facebook/wav2vec2-base")
model = Wav2Vec2ForPreTraining.from_pretrained("facebook/wav2vec2-base")

ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
input_values = feature_extractor(ds[0]["audio"]["array"], return_tensors="pt").input_values  # Batch size 1

# compute masked indices
batch_size, raw_sequence_length = input_values.shape
sequence_length = model._get_feat_extract_output_lengths(raw_sequence_length)
mask_time_indices = _compute_mask_indices((batch_size, sequence_length), mask_prob=0.2, mask_length=2)
mask_time_indices = torch.tensor(mask_time_indices, device=input_values.device, dtype=torch.long)

with torch.no_grad():
    outputs = model(input_values, mask_time_indices=mask_time_indices)

# compute cosine similarity between predicted (=projected_states) and target (=projected_quantized_states)
cosine_sim = torch.cosine_similarity(outputs.projected_states, outputs.projected_quantized_states, dim=-1)

# show that cosine similarity is much higher than random
cosine_sim[mask_time_indices.to(torch.bool)].mean() > 0.5

# for contrastive loss training model should be put into train mode
model = model.train()
loss = model(input_values, mask_time_indices=mask_time_indices).loss

print(loss)
```

And here adding the **sample_negative_indices** as target to calculate the loss  : 

```py
import torch
from transformers import AutoFeatureExtractor, Wav2Vec2ForPreTraining
from transformers.models.wav2vec2.modeling_wav2vec2 import _compute_mask_indices , _sample_negative_indices
from datasets import load_dataset

feature_extractor = AutoFeatureExtractor.from_pretrained("facebook/wav2vec2-base")
model = Wav2Vec2ForPreTraining.from_pretrained("facebook/wav2vec2-base")

ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
input_values = feature_extractor(ds[0]["audio"]["array"], return_tensors="pt").input_values  # Batch size 1

# compute masked indices and negatives inddices
batch_size, raw_sequence_length = input_values.shape

sequence_length = model._get_feat_extract_output_lengths(raw_sequence_length).item()
mask_time_indices = _compute_mask_indices((batch_size, sequence_length), mask_prob=0.2, mask_length=2)
sampled_negative_indices = _sample_negative_indices((batch_size, sequence_length), model.config.num_negatives, mask_time_indices)
mask_time_indices = torch.tensor(mask_time_indices,device=input_values.device, dtype=torch.long)
sampled_negative_indices = torch.tensor(sampled_negative_indices, dtype=torch.long)


with torch.no_grad():
  outputs = model(input_values, mask_time_indices=mask_time_indices)

# compute cosine similarity between predicted (=projected_states) and target (=projected_quantized_states)
cosine_sim = torch.cosine_similarity(outputs.projected_states, outputs.projected_quantized_states, dim=-1)

# show that cosine similarity is much higher than random
cosine_sim[mask_time_indices.to(torch.bool)].mean() > 0.5


# for contrastive loss training model should be put into train mode
model = model.train()
loss = model(input_values, mask_time_indices=mask_time_indices, sampled_negative_indices=sampled_negative_indices).loss
print(loss.item())
```

Reply:
@sanchit-gandhi

Reply:
Hey @abdouaziz - thanks for the code snippet! I now understand the problem we're trying to fix here 🤗

We require `sampled_negative_indices` (distractors) in order to compute the contrastive loss terms. Hence, we only compute the loss if `sampled_negative_indices` is passed to the model. Otherwise, we set all the loss terms to `None` in `Wav2Vec2ForPreTraining`:
https://github.com/huggingface/transformers/blob/693ba2cc79ef46e8323b35f791233ce94f433b5d/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L1480-L1481
So in the first example, `sampled_negative_indices` was omitted, meaning the loss terms were `None`. 

The fix you've implemented passes the `sampled_negative_indices` to the model, meaning the `contrastive_loss` term is computed!